plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/OTelTelemetryPlugin.java

```diff
@@ -59,8 +67,11 @@ public String getName() {
     }
 
     private Telemetry telemetry(TelemetrySettings telemetrySettings) {
-        return new OTelTelemetry(new OTelTracingTelemetry(OTelResourceProvider.get(telemetrySettings, settings)), new MetricsTelemetry() {
-        });
+        final OpenTelemetrySdk openTelemetry = OTelResourceProvider.get(telemetrySettings, settings);
+        return new OTelTelemetry(
+            new OTelTracingTelemetry(openTelemetry, () -> openTelemetry.getSdkTracerProvider().close()),
```

Comment:
Why do you need to pass this callbacks?

```suggestion
            new OTelTracingTelemetry(openTelemetry),
```